crates/sui-core/src/consensus_handler.rs

```diff
@@ -53,11 +62,75 @@ impl<T> ConsensusHandler<T> {
             checkpoint_service,
             transaction_manager,
             parent_sync_store,
+            low_scoring_authorities: scores_per_authority,
+            committee,
             metrics,
         }
     }
 }
 
+/// Updates list of authorities that are deemed to have low reputation scores by consensus
+/// these may be lagging behind the network, byzantine, or not reliably participating for any reason.
+/// We want to ensure that the remaining set of validators once we exclude the low scoring authorities
+/// is including enough stake for a quorum, at the very least. It is also possible that no authorities
+/// are particularly low scoring, in which case this will result in storing an empty list.
+fn update_low_scoring_authorities(
+    low_scoring_authorities: Arc<DashMap<AuthorityName, u64>>,
+    committee: Committee,
+    reputation_scores: ReputationScores,
+) {
+    if !reputation_scores.final_of_schedule {
+        return;
+    }
+    let mut score_list = vec![];
+    for val in reputation_scores.scores_per_authority.values() {
+        score_list.push(*val as f64);
+    }
+
+    let median = statistical::median(&score_list);
+    let mut deviations = vec![];
+    let mut abs_deviations = vec![];
+    for (i, _) in score_list.clone().iter().enumerate() {
+        deviations.push((score_list[i] - median) * -1.0);
+        abs_deviations.push((score_list[i] - median).abs());
+    }
+
+    // adjusted median absolute deviation
+    let mad = statistical::median(&abs_deviations) / 0.5;
+    let mut low_scoring = vec![];
+    let mut rest = vec![];
+    for (i, (a, _)) in reputation_scores.scores_per_authority.iter().enumerate() {
+        let temp = deviations[i] / mad;
+        if temp > 2.5 {
+            low_scoring.push(a);
+        } else {
+            rest.push(AuthorityName::from(a));
+        }
+    }
+
+    low_scoring_authorities.clear();
+
+    // make sure the rest have at least quorum
+    let remaining_stake = rest.iter().map(|a| committee.weight(a)).sum::<u64>();
+    let quorum_threshold = committee.threshold::<true>();
+    if remaining_stake < quorum_threshold {
```

Comment:
Should we disable it or can we check scores and remove the highest scoring of the low scores until we have a quorum?